src/main/java/org/snomed/snowstorm/core/data/services/AutoMerger.java

```diff
@@ -0,0 +1,542 @@
+package org.snomed.snowstorm.core.data.services;
+
+import org.snomed.snowstorm.core.data.domain.*;
+import org.snomed.snowstorm.rest.ControllerHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+import static org.snomed.snowstorm.config.Config.DEFAULT_LANGUAGE_DIALECTS;
+
+@Component
+public class AutoMerger {
+    private enum ComponentType {
+        Concept, Description, Relationship,
+        ClassAxiom, GciAxiom, LanguageReferenceSetMember
+    }
+
+    @Autowired
+    private ConceptService conceptService;
+
+    @Autowired
+    @Lazy
+    private ReferenceSetMemberService referenceSetMemberService;
+
+    /**
+     * Construct a new instance of Concept, which is a combination of both sourceConcept and targetConcept. The branch path
+     * on which targetConcept has been edited must be supplied to help identify which fields of targetConcept have actually changed.
+     *
+     * @param sourceConcept The state that will be the baseline for the new, auto-merged Concept.
+     * @param targetConcept Edits belonging to this Concept will be re-applied to the new, auto-merged Concept.
+     * @param branchPath    The branch path on which targetConcept has changed.
+     * @return A new instance of Concept which is a combination of both sourceConcept and targetConcept.
+     */
+    public Concept autoMerge(Concept sourceConcept, Concept targetConcept, String branchPath) {
+        // Data required for auto-merging.
+        joinAxiomReferenceSetMembers(targetConcept, branchPath);
```

Comment:
@dmcgihtsdo What about GCIs?

Reply:
@CoderMChu , GCIs are included in the `getComponentsChangedOnTargetNew` method.

Reply:
Don't they need to be loaded from refset members the same way as class axioms?

Reply:
Sorry, I never noticed you were talking about the `joinAxiomReferenceSetMembers` method @CoderMChu. Good catch. I'll update this method.